great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py

```diff
@@ -28,111 +30,248 @@ def plot(
+                    prescriptive=prescriptive,
+                )
+            )
+            charts.append(table_domain_chart)
+
+        return charts
+
+    def _create_chart_for_table_domain_expectation(
+        self,
+        expectation_configuration: ExpectationConfiguration,
+        attributed_metrics: Dict[Domain, Dict[str, ParameterNode]],
+        prescriptive: bool,
+    ) -> alt.Chart:
         attributed_values_by_metric_name: Dict[str, ParameterNode] = list(
-            attributed_metrics_by_domain.values()
+            attributed_metrics.values()
         )[0]
 
         # Altair does not accept periods.
-        # TODO: <Alex>ALEX Currently, only one Domain key (with domain_type of MetricDomainTypes.TABLE) is utilized; enhancements may require additional Domain key(s) with different domain_type value(s) to be incorporated.</Alex>
         metric_name: str = list(attributed_values_by_metric_name.keys())[0].replace(
             ".", "_"
         )
         domain_name: str = "batch"
-
-        domain_type: str = AltairDataTypes.ORDINAL.value
         metric_type: str = AltairDataTypes.QUANTITATIVE.value
+        domain_type: str = AltairDataTypes.ORDINAL.value
 
-        batch_ids: KeysView[str]
-        metric_values: MetricValues
-        batch_ids, metric_values = list(attributed_values_by_metric_name.values())[
-            0
-        ].keys(), sum(list(attributed_values_by_metric_name.values())[0].values(), [])
-
-        idx: int
-        batch_numbers: List[int] = [idx + 1 for idx in range(len(batch_ids))]
-
-        df: pd.DataFrame = pd.DataFrame(batch_numbers, columns=[domain_name])
-        df["batch_id"] = batch_ids
-        df[metric_name] = metric_values
+        df: pd.DataFrame = VolumeDataAssistantResult._create_df_for_charting(
+            metric_name, domain_name, attributed_values_by_metric_name
+        )
 
-        plot_impl: Callable
-        charts: List[alt.Chart] = []
+        plot_impl: Callable[
+            [
+                pd.DataFrame,
+                str,
+                alt.StandardType,
+                str,
+                alt.StandardType,
+            ],
+            alt.Chart,
+        ]
 
-        expectation_configurations: List[
-            ExpectationConfiguration
-        ] = self.expectation_suite.expectations
-        expectation_configuration: ExpectationConfiguration
         if prescriptive:
-            for expectation_configuration in expectation_configurations:
-                if (
-                    expectation_configuration.expectation_type
-                    == "expect_table_row_count_to_be_between"
-                ):
-                    for (
-                        kwarg_name,
-                        kwarg_value,
-                    ) in expectation_configuration.kwargs.items():
-                        df[kwarg_name] = kwarg_value
-
-            plot_impl = self._plot_prescriptive
+            for kwarg_name in ("min_value", "max_value"):
```

Comment:
Do we have to hard-code this or can we just use all `kwargs` like in the removed code?

Reply:
I thought not but after trying it out, it works as expected! changed